### PR TITLE
Test/Embed/Reschedule

### LINKED ISSUE #5862: [CAL-528] Playwright test for reschedule within embed
Add reschedule to testpage. Follow it up with a playwright test for reschedule.

_Originally posted by @hariombalhara in https://github.com/calcom/cal.com/pull/5861#discussion_r1039367452_
      

<sub>[CAL-528](https://linear.app/calcom/issue/CAL-528/playwright-test-for-reschedule-within-embed)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

closes #5862
Adds test for reschedule through embed

### CODE REVIEW

File: packages/embeds/embed-core/index.html [hunk 1]
```diff
@@ -97,7 +97,10 @@ <h2>Popup Examples</h2>
       <button data-cal-namespace="popupTeamLinkLightTheme" data-cal-config='{"theme":"light"}' data-cal-link="team/seeded-team/collective-seeded-team-event">Book with Test Team[Light Theme]</button>
       <button data-cal-namespace="popupTeamLinkDarkTheme" data-cal-config='{"theme":"dark"}' data-cal-link="team/seeded-team/collective-seeded-team-event">Book with Test Team[Dark Theme]</button>
       <button data-cal-namespace="popupTeamLinksList" data-cal-link="team/seeded-team/">See Team Links [Auto Theme]</button>
-      <button data-cal-namespace="popupReschedule" data-cal-link="reschedule/qm3kwt3aTnVD7vmP9tiT2f">Reschedule Event[Auto Theme]</button>
+      <script>
+        let popupRescheduleId = new URL(document.URL).searchParams.get("popupRescheduleId") || "qm3kwt3aTnVD7vmP9tiT2f"
+        document.write(`<button data-cal-namespace="popupReschedule" data-cal-link="reschedule/${popupRescheduleId}">Reschedule Event[Auto Theme]</button>`)
```
Comment (hariombalhara) [inline packages/embeds/embed-core/index.html:102]: Allows, playwright tests to dynamically use a bookingId to reschedule

File: packages/embeds/embed-core/playwright/config/playwright.config.ts [hunk 2]
```diff
@@ -150,6 +150,8 @@ expect.extend({
       }, 500);
     });
 
+    clearInterval(iframeReadyCheckInterval);
```
Comment (hariombalhara) [inline packages/embeds/embed-core/playwright/config/playwright.config.ts:153]: It was causing `getActionFiredDetails` to be unnecessarily called in timer when it wasn't needed.

File: packages/embeds/embed-core/playwright/lib/testUtils.ts [hunk 3]
```diff
@@ -33,31 +35,41 @@ export const getBooking = async (bookingId: string) => {
...
-      if (!iframe) {
-        resolve(false);
-        return;
-      }
       const interval = setInterval(() => {
+        const iframe = document.querySelector(".cal-embed") as HTMLIFrameElement | null;
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         // @ts-ignore
-        if (iframe.contentWindow && window.iframeReady) {
+        if (iframe && iframe.contentWindow && window.iframeReady) {
```
Comment (hariombalhara) [inline packages/embeds/embed-core/playwright/lib/testUtils.ts:44]: Make the check more reliable

File: packages/embeds/embed-core/playwright/lib/testUtils.ts [hunk 4]
```diff
@@ -104,10 +116,35 @@ export async function bookFirstEvent(username: string, frame: Frame, page: Page)
   await frame.fill('[name="email"]', "embed-user@example.com");
   await frame.press('[name="email"]', "Enter");
   const response = await page.waitForResponse("**/api/book/event");
-  const responseObj = await response.json();
-  const bookingId = responseObj.uid;
+  const booking = (await response.json()) as { uid: string; eventSlug: string };
+  booking.eventSlug = eventSlug;
```
Comment (hariombalhara) [inline packages/embeds/embed-core/playwright/lib/testUtils.ts:120]: Useful for reschedule test to know, which `eventSlug` would `reschedule/{UID}` would get redirected to.

**Review by hariombalhara — COMMENTED** [review]
Self review done

**Review by PeerRich — APPROVED** [review]

**Pull request merged**